File: JAVA/ATM_Design/src/main/java/org/ceo/states/EjectingCardState.java
package org.ceo.states;

import org.ceo.enums.ATMState;
import org.ceo.models.ATM;

public class EjectingCardState extends ATMStateMachine{
    private final ATM atm;

    public EjectingCardState(ATM atm) {
        this.atm = atm;
    }

    @Override
    public void ejectCard() {
        System.out.println("Ejected Card Successfully");
        this.atm.changeState(new ReadyForTransactionState(this.atm));
    }

    @Override
    public ATMState getState() {
        return ATMState.EJECTING_CARD;
    }
}

File: C++/ATM-System/src/states/EjectingCardState.cpp
#include "../states/EjectingCardState.h"
#include "../models/ATM.h"
#include "../states/ReadyForTransactionState.h"
#include <iostream>
#include <stdexcept>

EjectingCardState::EjectingCardState(ATM* atm) : atm(atm) {}



void EjectingCardState::ejectCard() {
    std::cout << "Ejected Card Successfully" << std::endl;
    atm->changeState(new ReadyForTransactionState(atm));
}


ATMState EjectingCardState::getState() {
    return ATMState::EJECTING_CARD;
}
